src/components.jl

```diff
@@ -648,75 +657,41 @@ function _relative_position(xmin, xmax, pos::Length{:pct}, scale::Symbol)
     end
 end
 
+# annotation coordinates in pct
 const position_multiplier = Dict(
-    :N            => (0.5pct, 0.9pct),
-    :NE           => (0.9pct, 0.9pct),
-    :E            => (0.9pct, 0.5pct),
-    :SE           => (0.9pct, 0.1pct),
-    :S            => (0.5pct, 0.1pct),
-    :SW           => (0.1pct, 0.1pct),
-    :W            => (0.1pct, 0.5pct),
-    :NW           => (0.1pct, 0.9pct),
-    :topleft      => (0.1pct, 0.9pct),
-    :topcenter    => (0.5pct, 0.9pct),
-    :topright     => (0.9pct, 0.9pct),
-    :bottomleft   => (0.1pct, 0.1pct),
-    :bottomcenter => (0.5pct, 0.1pct),
-    :bottomright  => (0.9pct, 0.1pct),
+    :N            => (0.5, 0.9),
+    :NE           => (0.9, 0.9),
+    :E            => (0.9, 0.5),
+    :SE           => (0.9, 0.1),
+    :S            => (0.5, 0.1),
+    :SW           => (0.1, 0.1),
+    :W            => (0.1, 0.5),
+    :NW           => (0.1, 0.9),
+    :topleft      => (0.1, 0.9),
+    :topcenter    => (0.5, 0.9),
+    :topright     => (0.9, 0.9),
+    :bottomleft   => (0.1, 0.1),
+    :bottomcenter => (0.5, 0.1),
+    :bottomright  => (0.9, 0.1),
 )
 
 # Give each annotation coordinates based on specified position
-function locate_annotation(sp::Subplot, pos::Symbol, label::PlotText)
-    x, y = position_multiplier[pos]
-    (
-        _relative_position(
-            axis_limits(sp, :x)...,
-            x,
-            sp[get_attr_symbol(:x, :axis)][:scale],
-        ),
+locate_annotation(sp::Subplot, rel::Tuple, label::PlotText) = (
+    map(1:length(rel), [:x, :y, :z]) do i, letter
```

Comment:
```suggestion
    map(1:length(rel), (:x, :y, :z)) do i, letter
```

Use `Tuple`s where you can (statically sized).

Reply:
Good point!